### PR TITLE
Allow dynamic shapes inference in GPU

### PR DESCRIPTION
**Pull request by sspintel**
### Description

- Allow dynamic shapes inference in GPU when enable_dynamic_shapes=True is set (Lines of interest: [Link 1](https://github.com/intel/onnxruntime/pull/290/files#diff-84b5dc31f64204b79173c207589103341ae50d58779a3e03092a6eb9b8e5cb7fL266) [Link 2](https://github.com/intel/onnxruntime/pull/290/files#diff-84b5dc31f64204b79173c207589103341ae50d58779a3e03092a6eb9b8e5cb7fR80) [Link 3](https://github.com/intel/onnxruntime/pull/290/files#diff-10ceae516462e95d43795fa0eb5ac08caa78e2e0ed0ed7491347ee840f8ca765R165)
- Add Synchronous inference API Infer(); Useful for debugging purposes (Files of interest: ov_interface.*)
- Added formatting changes in the files involved (Rest of the lines)

### CODE REVIEW

**Review by MaajidKhan — CHANGES_REQUESTED** [review]

File: onnxruntime/core/providers/openvino/ov_interface.cc [hunk 1]
```diff
@@ -108,6 +108,16 @@ namespace onnxruntime {
         }
     }
 
+    void OVInferRequest::Infer() {
```
[the remark this answers is not in the document]
Comment (sspintel) [reply onnxruntime/core/providers/openvino/ov_interface.cc:111]: Yes it's not being used now, but debugging into OpenVINO code during async inference calls is difficult as we have to track down which thread the inference is actually running on. Instead, when debugging I just replace StartAsync with Infer and remove Wait.

File: onnxruntime/core/providers/openvino/backend_manager.cc [hunk 2]
```diff
@@ -75,37 +75,33 @@ BackendManager::BackendManager(const onnxruntime::Node& fused_node,
...
+                           << "Backend created for graph " << subgraph_context_.subgraph_name;
       }
     }
-  } else if (ModelHasSymbolicInputDims(subgraph) &&
-      GetGlobalContext().device_type.find("GPU") != std::string::npos) {
-    LOGS_DEFAULT(INFO) << "[OpenVINO-EP] Model has symbolic input dims. Defering backend initialization";
-    subgraph_context_.has_dynamic_input_shape = true;
   } else {
-    LOGS_DEFAULT(INFO) << "[OpenVINO-EP] Model has concreate input dims. Initializing backend for graph " << subgraph_context_.subgraph_name;
+    LOGS_DEFAULT(INFO) << "[OpenVINO-EP] Model has concrerte input dims. Initializing backend for graph " << subgraph_context_.subgraph_name;
```
[the remark this answers is not in the document]
Comment (sspintel) [reply onnxruntime/core/providers/openvino/backend_manager.cc]: Fixed typo

File: onnxruntime/core/providers/openvino/backend_manager.cc [hunk 3]
```diff
@@ -75,37 +75,33 @@ BackendManager::BackendManager(const onnxruntime::Node& fused_node,
...
-          try {
-            concrete_backend_ = BackendFactory::MakeBackend(*model_proto_,
-                                                            GetGlobalContext(),
-                                                            subgraph_context_);
-          } catch (std::string const & msg) {
-              throw msg;
-          }
+                           << "Creating backend Dynamic Shapes";
+        try {
+          concrete_backend_ = BackendFactory::MakeBackend(*model_proto_,
```
[the remark this answers is not in the document]
Comment (sspintel) [reply onnxruntime/core/providers/openvino/backend_manager.cc:85]: I think we can keep these variables as-is Maajid. It's a bit confusing if we think about tensor shapes (static vs dynamic) but here concrete and dynamic seem to refer to whether backends get created once during initialization (for static inputs or for backends supporting dynamic shapes) or for every unique shape for backends that don't.

**Comment by MaajidKhan** [comment]
Thanks SP. PR is now merged.

**Pull request merged**